src/components/Description.svelte

```diff
@@ -1,21 +1,46 @@
 <script lang="ts">
-  import { onMount } from 'svelte'
-  import { fetchQuestion } from '../fetch'
-  import type { Context } from '../types'
-  import type { editor } from 'monaco-editor'
+  import { afterUpdate, onMount } from "svelte";
+  import { fetchQuestion } from "../fetch";
+  import type { Context } from "../types";
+  import type { editor } from "monaco-editor";
+  import { locale } from "../stores";
+  import { markdownToHtml } from "../markdown";
+  import { findNearlyAnchorElement } from "../dom";
 
-  export let context: Context
+  export let context: Context;
+  export let quiz: number;
+  export let toNext: (quiz: number) => void;
 
-  let rendered: string | null = null
-  let markersLoaded = false
-  let markers: editor.IMarker[] = []
+  let root: HTMLElement;
+  let markers: editor.IMarker[] = [];
+  let playgroundUrl: string = "";
+  let frame: HTMLIFrameElement;
+  let loadPromise: Promise<string>;
+  let isMarkerLoaded = false;
+  $: isPassed = isMarkerLoaded && markers.length === 0;
 
   onMount(async () => {
-    rendered = await fetchQuestion(context)
+    loadPromise = onSwitchQuestion(quiz, $locale);
+  });
 
-    // set a delay for letting TypeScript does type-checking
-    setTimeout(() => (markersLoaded = true), 3000)
-  })
+  $: if (quiz) {
+    loadPromise = onSwitchQuestion(quiz, $locale);
+  }
+
+  async function onSwitchQuestion(quiz: number, locale: string) {
+    if (!isNaN(quiz)) {
+      const { readme, playUrl } = await fetchQuestion(quiz, locale);
+      if (playUrl) {
+        isMarkerLoaded = false;
+        playgroundUrl = playUrl;
+        setTimeout(() => {
+          isMarkerLoaded = true;
+        }, 2000);
+        return await markdownToHtml(readme, context.sandbox);
```

Comment:
We aren't in `try`...`catch` block, so we can simplify like this:

```suggestion
        return markdownToHtml(readme, context.sandbox);
```

Reply:
There need to wait for code colorize to complete

https://github.com/type-challenges/playground-plugin/blob/b8dc614740784ae956523a30a744ccf4a8a6ea10/src/markdown.ts#L30-L41

Reply:
I know, but here it just returns the result of `markdownToHtml`, so nothing different. Anyway, this isn't blocker.